quality_control/models/qc_inspection.py

```diff
@@ -0,0 +1,299 @@
+         ('failed', 'Quality failed'),
+         ('canceled', 'Cancelled')],
+        string='State', readonly=True, default='draft')
+    success = fields.Boolean(
+        compute="_success", string='Success',
+        help='This field will be marked if all tests have been succeeded.',
+        store=True)
+    auto_generated = fields.Boolean(
+        string='Auto-generated', readonly=True, copy=False,
+        help='If an inspection is auto-generated, it can be cancelled nor '
+             'removed')
+    company_id = fields.Many2one(
+        comodel_name='res.company', string='Company', readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: self.env['res.company']._company_default_get(
+            'qc.inspection'))
+    user = fields.Many2one(
+        comodel_name='res.users', string='Responsible',
+        track_visibility='always', default=lambda self: self.env.user)
+
+    @api.model
+    def create(self, vals):
+        if vals.get('name', '/') == '/':
+            vals['name'] = self.env['ir.sequence'].get('qc.inspection')
+        return super(QcInspection, self).create(vals)
+
+    @api.multi
+    def unlink(self):
+        for inspection in self:
+            if inspection.auto_generated:
+                raise exceptions.Warning(
+                    _("You cannot remove an auto-generated inspection"))
+            if inspection.state != 'draft':
+                raise exceptions.Warning(
+                    _("You cannot remove an inspection that it's not in draft "
+                      "state"))
+        return super(QcInspection, self).unlink()
+
+    @api.multi
+    def action_draft(self):
+        self.write({'state': 'draft'})
+
+    @api.multi
+    def action_todo(self):
+        for inspection in self:
+            if not inspection.test:
+                raise exceptions.Warning(
+                    _("You must set the test to perform first."))
+        self.write({'state': 'ready'})
+
+    @api.multi
+    def action_confirm(self):
+        for inspection in self:
+            for line in inspection.inspection_lines:
+                if line.question_type == 'qualitative':
+                    if not line.qualitative_value:
+                        raise exceptions.Warning(
+                            _("You should provide an answer for all "
+                              "quantitative questions."))
+                else:
+                    if not line.uom_id:
+                        raise exceptions.Warning(
+                            _("You should provide a unit of measure for "
+                              "qualitative questions."))
+            if inspection.success:
+                inspection.state = 'success'
+            else:
+                inspection.state = 'waiting'
+
+    @api.multi
+    def action_approve(self):
+        for inspection in self:
+            if inspection.success:
+                inspection.state = 'success'
```

Comment:
This change implies too much (possible errors replacing code+data+views, migration scripts...) for the worth.